tests/ops/test_qubit_ops.py

```diff
@@ -767,6 +768,112 @@ def test_qubit_unitary_exceptions(self):
         with pytest.raises(ValueError, match="must be unitary"):
             qml.QubitUnitary(U3, wires=0).matrix
 
+    X = np.array([[0, 1], [1, 0]])
+
+    def test_controlled_qubit_unitary(self):
+        """Test if ControlledQubitUnitary returns the correct matrix for a control-control-X
+        (Toffoli) gate"""
+        mat = qml.ControlledQubitUnitary(X, control_wires=[0, 1], wires=2).matrix
+        mat2 = qml.Toffoli(wires=[0, 1, 2]).matrix
+        assert np.allclose(mat, mat2)
+
+    def test_controlled_qubit_unitary_no_control(self):
+        """Test if ControlledQubitUnitary raises an error if control wires are not specified"""
+        with pytest.raises(ValueError, match="Must specify control wires"):
+            qml.ControlledQubitUnitary(X, wires=2)
+
+    def test_controlled_qubit_unitary_shared_control(self):
+        """Test if ControlledQubitUnitary raises an error if control wires are shared with wires"""
+        with pytest.raises(ValueError, match="The control wires must be different from the wires"):
+            qml.ControlledQubitUnitary(X, control_wires=[0, 2], wires=2)
+
+    @pytest.mark.parametrize("target_wire", range(3))
+    def test_controlled_qubit_unitary_toffoli(self, target_wire):
+        """Test if ControlledQubitUnitary acts like a Toffoli gate when the input unitary is a
+        single-qubit X. This test allows the target wire to be any of the three wires."""
```

Comment:
Shouldn't this _not_ be the case? If applying the unitary to e.g. the second qubit, with control on the first should be:
```
[[1, 0, 0, 0],
 [0, 1, 0, 0],
 [0, 0, 0, 1],
 [0, 0, 1, 0]]
```

while applying the unitary to the first with control on the second should be
```
[[1, 0, 0, 0],
 [0, 0, 0, 1],
 [0, 0, 1, 0],
 [0, 1, 0, 0]]
```
Am I missing something here? It seems like `ControlledQubitUnitary` doesn't care which wires are control/applied.

Reply:
> This test allows the target wire to be any of the three wires

I.e., this means that the target wire can be any of the three, which results in a different transformation, and the test checks if the correct transformation has been performed (by comparing against the action of `Toffoli`).

Reply:
Thanks! This explains it as well. :+1: